okhttp-testing-support/src/main/java/okhttp3/OkHttpClientTestRule.kt

```diff
@@ -119,6 +133,19 @@ class OkHttpClientTestRule : TestRule {
     }
   }
 
+  /** Returns true if this queue became idle before the timeout elapsed. */
+  private fun TaskQueue.awaitIdle(timeout: Long, timeUnit: TimeUnit): Boolean {
+    val latch = CountDownLatch(1)
+    schedule(object : Task("awaitIdle") {
```

Comment:
Making a Task companion factory would be nice here.

```
Task("awaitIdle") {
```

Reply:
I tried it and it doesn’t come out as satisfying as I was hoping because of the return value.

```
    schedule(Task("awaitIdle") {
      latch.countDown()
      return@Task -1L
    })
```

Lemme convert a few more of our thread pools over to use this new framework and then I’d like to take another attempt at tightening up the syntax.